add the new vstring API to Devel::PPPort

tonycoz opened the pull request:
<!--
A good description should explain the problem the pull request addresses
and give context to the reviewers to aid them in their reviews.
-->

Adds the new vstring APIs to ppport.h

Also improves a Devel::PPPort test and extends `Porting/test-dist-modules.pl` to aid in development.

<!--
Significant changes to Perl must be documented in perldelta.

Consider if the changes in this pull request are worthy of a perldelta
entry, then pick the appropriate line below and remove the others.
-->
---------------------------------------------------------------------------------
* This set of changes does not require a perldelta entry.

tonycoz committed:
Devel-PPPort: improve ppptest test reporting

A change I made caused this to fail, tracking it down was rough.

tonycoz committed:
test-dist-modules.pl: allow the invoker to keep the build around

This is especially handy when tracking down problems with
Devel::PPPort, since so much is generated.

tonycoz committed:
Devel-PPPort: add SvVSTRING()

vstrings were originally added in perl-5.8.0-82-g92f0c26562,
SvVSTRING_mg() was originally added in perl-5.8.0-8018-gb0a11fe104 so
technically in the same release.

I expect there were some uses of intermediate versions 19 years ago,
but I don't think we need to worry about it now.

tonycoz committed:
Storable: use SvVSTRING() from ppport.h

Leont commented on dist/Devel-PPPort/parts/inc/magic:
```diff
@@ -725,6 +773,13 @@ if (ivers($]) >= ivers("5.6")) {
 is tied($big)->{fetch}, 1;
 is tied($big)->{store}, 0;
 
+SKIP:
```
Why this `SKIP:` here?

Leont commented:
I think it also needs to be added to `parts/`, otherwise the `ppport.h` script will not detect that an XS module using `SvVSTRING`/`sv_vstring_get` will need to define `NEED_sv_vstring_get`.

I've never been 100% sure of how that part works, but I suspect it needs to be set to `#T` in `parts/todo/5003007`, and ti `#U# in `parts/base/5041010`.

tonycoz committed:
Devel::PPPort: add compatibility entries for the new vstrings

Based on @leont's comment on #23160.

This seems to produce reasonable results:

tony@venus:.../git/perl6$ cat foo.c
SvVSTRING
tony@venus:.../git/perl6$ ./perl -Ilib dist/Devel-PPPort/ppport.h --nofilter foo.c
Scanning foo.c ...
=== Analyzing foo.c ===
Uses SvVSTRING, which depends on sv_vstring_get, SvVSTRING_mg, mg_find, PERL_MAGIC_vstring, SvMAGICAL
File needs sv_vstring_get, adding static request
Needs to include 'ppport.h'
Analysis completed
Suggested changes:
--- foo.c       2025-04-01 10:51:39.040415623 +1100
+++ foo.c.patched       2025-04-01 10:55:11.347014468 +1100
@@ -1 +1,3 @@
+#define NEED_sv_vstring_get
+#include "ppport.h"
 SvVSTRING

Leont reviewed (approved):
It looks good to me (though I would squash the fixes)
